deeplake/util/casting.py

```diff
@@ -34,7 +20,7 @@ def get_dtype(val: Union[np.ndarray, Sequence, Sample]) -> np.dtype:
     elif isinstance(val, bool):
         return np.dtype(bool)
     elif isinstance(val, Sequence):
-        return reduce(_get_bigger_dtype, map(get_dtype, val))
+        return get_dtype(next(iter(val)))
```

Comment:
Dont make any changes in this function. Instead in `chunk_engine.py::_sanitize_samples`,  do `tensor_meta.set_dtype(get_dtype(samples[0]))`